src/overlay/OverlayManagerImpl.cpp

```diff
@@ -257,9 +261,43 @@ OverlayManagerImpl::start()
                 this->tick();
             },
             VirtualTimer::onFailureNoop);
+
+        mPeerIPTimer.async_wait([this]() { this->maybeRefreshPeerIPs(); },
+                                VirtualTimer::onFailureNoop);
     }
 }
 
+PeerBareAddress
+OverlayManagerImpl::getLatestPeerAddress(std::string const& peerStr,
```

Comment:
That name is misleading, even if this method return the lastest address.
Problem is that is also updates peer cache, updates peer table and possibly updates mConfigurationPreferredPeers, so it is much more than getting that address.

My proposition is to make it responsible for less things.

1. isPreferred parameter could be removed, and updating mConfigurationPreferredPeers could be done in maybeRefreshPeerIPs
2. name could be changed to something like updatePeerAddressCache
3. call to addAddressToExistingPeer could also be moved to maybeRefreshPeerIPs, and condition if (storedAddress != address) should be added to PeerManager::addAddressToExistingPeer

Reply:
What do you think about instead renaming this function to something like `resolveIPAndMaybeUpdateCaches`? The reason I don't like splitting it is because all cache/db updates are consolidated in one place, so it's less overhead to track these. Also, `storedAddress != address` condition prevents from redundant cache inserts.

Reply:
I think that with method's name so long it may be better to extract small utility class just for cache handling (if the cache will be still there, after resolving Nicolas comments).